hugegraph-test/src/main/java/com/baidu/hugegraph/unit/rocksdb/RocksDBSessionTest.java

```diff
@@ -0,0 +1,533 @@
+        this.commit();
+
+        Assert.assertArrayEquals(null, session.get(TABLE, key11));
+        Assert.assertArrayEquals(null, session.get(TABLE, key12));
+        Assert.assertArrayEquals(value21, session.get(TABLE, key21));
+    }
+
+    @Test
+    public void testDeleteByRangeWithMinMaxByteValue() throws RocksDBException {
+        Session session = this.rocks.session();
+
+        byte[] key11 = new byte[]{1, 0};
+        byte[] value11 = b("value-1-1");
+        session.put(TABLE, key11, value11);
+
+        byte[] key12 = new byte[]{1, 127};
+        byte[] value12 = b("value-1-2");
+        session.put(TABLE, key12, value12);
+
+        byte[] key13 = new byte[]{1, (byte) 0x80}; // 128
+        byte[] value13 = b("value-1-3");
+        session.put(TABLE, key13, value13);
+
+        byte[] key14 = new byte[]{1, (byte) 0xff}; // 255
+        byte[] value14 = b("value-1-4");
+        session.put(TABLE, key14, value14);
+
+        byte[] key20 = new byte[]{2, 0};
+        byte[] value20 = b("value-2-0");
+        session.put(TABLE, key20, value20);
+
+        session.deleteRange(TABLE,
+                            new byte[]{1, 0}, new byte[]{1, (byte) 0xff});
+        this.commit();
+
+        Assert.assertArrayEquals(null, session.get(TABLE, key11));
+        Assert.assertArrayEquals(null, session.get(TABLE, key12));
+        Assert.assertArrayEquals(null, session.get(TABLE, key13));
+        Assert.assertArrayEquals(value14, session.get(TABLE, key14));
+        Assert.assertArrayEquals(value20, session.get(TABLE, key20));
+
+        session.deleteRange(TABLE,
+                            new byte[]{1, (byte) 0xff}, new byte[]{2, 0});
+        this.commit();
+
+        Assert.assertArrayEquals(null, session.get(TABLE, key11));
+        Assert.assertArrayEquals(null, session.get(TABLE, key12));
+        Assert.assertArrayEquals(null, session.get(TABLE, key13));
+        Assert.assertArrayEquals(null, session.get(TABLE, key14));
+        Assert.assertArrayEquals(value20, session.get(TABLE, key20));
+    }
+
+    @Test
+    public void testDeleteSingle() throws RocksDBException {
+        put("person:1gname", "James");
+        put("person:1gage", "19");
+        put("person:1gname", "James2");
+
+        Assert.assertEquals("James2", get("person:1gname"));
+        Assert.assertEquals("19", get("person:1gage"));
+
+        // deleteSingle after put once
+        this.rocks.session().deleteSingle(TABLE, b("person:1gage"));
+        this.commit();
+
+        Assert.assertEquals("James2", get("person:1gname"));
+        Assert.assertEquals(null, get("person:1gage"));
+
+        // deleteSingle after put twice
+        this.rocks.session().deleteSingle(TABLE, b("person:1gname"));
+        this.commit();
+
+        // NOTE: maybe return "James" here
+        Assert.assertEquals(null, get("person:1gname"));
+        Assert.assertEquals(null, get("person:1gage"));
+    }
+
+    @Test
+    public void testCompact() throws RocksDBException {
```

Comment:
seems no obvious signal for compacted

Reply:
compactRange() is a synchronous blocked call. there is no way to verify whether the compaction is done, just can be tested in terms of performance.

Reply:
compactRange() is a synchronous blocked call. there is no way to verify whether the compaction is done, just can be tested in terms of performance.

Reply:
compactRange() is a synchronous blocked call. there is no way to verify whether the compaction is done, just can be tested in terms of performance.